modules/perception/onboard/transform_wrapper/transform_wrapper.cc

```diff
@@ -157,18 +157,16 @@ bool TransformWrapper::GetSensor2worldTrans(
   trans_novatel2world.timestamp = timestamp;
   Eigen::Affine3d novatel2world;
 
-  if (QueryTrans(timestamp, &trans_novatel2world, novatel2world_tf2_frame_id_,
-                 novatel2world_tf2_child_frame_id_) != true) {
-    if (FLAGS_obs_enable_local_pose_extrapolation) {
-      if (!transform_cache_.QueryTransform(
-              timestamp, &trans_novatel2world,
+  if (!QueryTrans(timestamp, &trans_novatel2world, novatel2world_tf2_frame_id_,
+                 novatel2world_tf2_child_frame_id_)) {
+    if (!FLAGS_obs_enable_local_pose_extrapolation ||
+        !transform_cache_.QueryTransform(timestamp, &trans_novatel2world,
               FLAGS_obs_max_local_pose_extrapolation_latency)) {
         return false;
-      }
-    } else {
-      return false;
     }
-  } else if (FLAGS_obs_enable_local_pose_extrapolation) {
+  }
+
+  if (FLAGS_obs_enable_local_pose_extrapolation) {
     transform_cache_.AddTransform(trans_novatel2world);
   }
 
```

Comment:
The two logics are different. Better check how the function QueryTransform is implemented.

Reply:
Agree.
(QueryTrans=false, FLAGS_obs_enable_local_pose_extrapolation=true) will result in different execution.

Reply:
Thanks for both your comments! @javeme yep, let's check how to make the code looks a better gracefully.